docs/writing-rules/testing-rules.md

```diff
@@ -69,82 +93,116 @@ eval(safe_get_user_input())
 ```sh
-1 yaml files tested
-check id scoring:
---------------------------------------------------------------------------------
-(TODO: 2) rules/detect-eval.yaml
-	✖ insecure-eval-use                                  TP: 1 TN: 2 FP: 1 FN: 1
-	test: rules/detect-eval.py, expected lines: [5, 12], reported lines: [5, 15]
---------------------------------------------------------------------------------
-final confusion matrix: TP: 1 TN: 2 FP: 1 FN: 1
---------------------------------------------------------------------------------
+1/1: ✓ All tests passed
+No tests for fixes found.
 ```
 
-- True positives (`TP`) correspond to `ruleid`
-- True negatives (`TN`) correspond to `ok`
-- False positives (`FP`) correspond to `todook`
-- False negatives (`FN`) correspond to `todoruleid`
+Semgrep tests automatically avoid failing on lines marked with `# todoruleid` or `# todook`.
+## Storing rules and test targets in different directories
 
-To avoid failing on TODOs you can specify `--test-ignore-todo`:
+Creating different directories for rules and tests helps users manage a growing library of custom rules.
+To store rules and test targets in different directories use the `--config` option.
+For the following directory structure,
 
 ```sh
-python -m semgrep --quiet --test --test-ignore-todo rules/
+$ tree tests
+
+tests
+├── rules
+│   └── python
+│       └── insecure-eval-use.yaml
+└── targets
+    └── python
+        └── insecure-eval-use.py
+
+4 directories, 2 files
 ```
 
-This will produce the following output:
+the command
+
 ```sh
-1 yaml files tested
-check id scoring:
---------------------------------------------------------------------------------
-(TODO: 2) rules/detect-eval.yaml
-	✔ insecure-eval-use                                  TP: 1 TN: 1 FP: 0 FN: 0
---------------------------------------------------------------------------------
-final confusion matrix: TP: 1 TN: 1 FP: 0 FN: 0
---------------------------------------------------------------------------------
+semgrep --test --config tests/rules/ tests/targets/
 ```
 
-To store rules and test targets in different directories you can specify `--config`:
+produces the same output as in the previous example.
 
-```sh
-tree tests
+The subdirectory structure of these two directories must be the same for Semgrep to correctly find the associated files.
+
+To test the autofix behavior, add the autofix test file `rules/detect-eval.fixed.py` to represent the expected outcome of applying the fix to the test code:
+
+```python
+from lib import get_user_input, safe_get_user_input, secure_eval
+
+user_input = get_user_input()
+# ruleid: insecure-eval-use
+secure_eval(user_input)
+
+# ok: insecure-eval-use
+eval('print("Hardcoded eval")')
+
+totally_safe_eval = eval
+# todoruleid: insecure-eval-use
+totally_safe_eval(user_input)
+
+# todook: insecure-eval-use
+secure_eval(safe_get_user_input())
 ```
 
-will produce the following output:
+So that the directory structure looks like this:
```

Comment:
Not sure if rendered is the best word as it may be associated with the process of rendering and different technical terms. Feel free to substitute this word.